applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.cpp

```diff
@@ -0,0 +1,1319 @@
+
+            // Compute constitutive law variables
+            SetConstitutiveVariables(this_kinematic_variables,  this_constitutive_variables, cl_values, point_number, r_integration_points);
+
+            // Call the constitutive law to update material variables
+            mConstitutiveLawVector[point_number]->FinalizeMaterialResponse(cl_values, ConstitutiveLaw::StressMeasure_Cauchy);
+        }
+    }
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SmallDisplacementMixedStrainDisplacementElement::CalculateLocalSystem(
+    MatrixType& rLHS,
+    VectorType& rRHS,
+    const ProcessInfo& rProcessInfo)
+{
+    KRATOS_TRY
+
+    const auto& r_geometry = GetGeometry();
+    auto &r_props          = GetProperties();
+    const SizeType dim     = r_geometry.WorkingSpaceDimension();
+    const SizeType n_nodes = r_geometry.PointsNumber();
+    const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+    const SizeType block_size  = dim + strain_size;
+    const SizeType matrix_size = block_size * n_nodes;
+    const double tau = GetStabilizationFactor();
+
+    // Check RHS size
+    if (rRHS.size() != matrix_size) {
+        rRHS.resize(matrix_size, false);
+    }
+
+    // Check LHS size
+    if (rLHS.size1() != matrix_size || rLHS.size2() != matrix_size) {
+        rLHS.resize(matrix_size, matrix_size, false);
+    }
+    noalias(rLHS) = ZeroMatrix(matrix_size, matrix_size);
+    noalias(rRHS) = ZeroVector(matrix_size);
+
+    // Create the kinematics container and fill the nodal data
+    KinematicVariables kinematic_variables(strain_size, dim, n_nodes);
+
+    // Compute U and E
+    GetNodalDoFsVectors(kinematic_variables.NodalDisplacements, kinematic_variables.NodalStrains);
+
+    // Create the constitutive variables and values containers
+    ConstitutiveVariables constitutive_variables(strain_size);
+    ConstitutiveLaw::Parameters cons_law_values(r_geometry, r_props, rProcessInfo);
+    auto& r_cl_options = cons_law_values.GetOptions();
+    r_cl_options.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+    r_cl_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    r_cl_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+
+    cons_law_values.SetStrainVector(constitutive_variables.StrainVector);
+    cons_law_values.SetStressVector(constitutive_variables.StrainVector);
+    cons_law_values.SetConstitutiveMatrix(constitutive_variables.D);
+
+    Vector RHSu(dim * n_nodes), RHSe(strain_size * n_nodes);
+    noalias(RHSu) = ZeroVector(dim * n_nodes);
+    noalias(RHSe) = ZeroVector(strain_size * n_nodes);
+
+    Matrix K(dim * n_nodes, dim * n_nodes), G(n_nodes * strain_size, dim * n_nodes),
+        M(n_nodes * strain_size, n_nodes * strain_size), Q(dim * n_nodes, n_nodes * strain_size);
```

Comment:
```suggestion
    M(n_nodes * strain_size, n_nodes * strain_size), Q(dim * n_nodes, n_nodes * strain_size);
```
Indentation (me da flojera)